packages/lisk-p2p/test/unit/errors.ts

```diff
@@ -83,4 +87,74 @@ describe('errors', () => {
 			});
 		});
 	});
+
+	describe('#RPCGetPeersFailed', () => {
+		let rpcGetPeersFailed: RPCGetPeersFailedError;
+		const peerId = '127.0.0.1:5001';
+		const defaultMessage = `Error when fetching peerlist of peer with peer id ${peerId}`;
+		const defaultError = new Error('Peer not available');
+
+		beforeEach(() => {
+			rpcGetPeersFailed = new RPCGetPeersFailedError(
+				defaultMessage,
+				defaultError,
+				peerId,
+			);
+			return Promise.resolve();
+		});
+
+		describe('should create an error object', () => {
+			it('should create a new instance of RPCGetPeersFailed', () => {
+				return expect(rpcGetPeersFailed)
+					.to.be.an('object')
+					.and.be.instanceof(RPCGetPeersFailedError);
+			});
+
+			it('should set error name to `RPCGetPeersFailed`', () => {
+				return expect(rpcGetPeersFailed.name).to.eql('RPCGetPeersFailed');
+			});
+		});
+
+		describe('should set error object properties', () => {
+			beforeEach(() => {
+				rpcGetPeersFailed = new RPCGetPeersFailedError(
+					defaultMessage,
+					defaultError,
+					peerId,
+				);
+				return Promise.resolve();
+			});
+
+			it('should set error property info when passed as an argument', () => {
+				return expect(rpcGetPeersFailed)
+					.and.to.have.property('peerId')
+					.which.is.eql(peerId);
+			});
+
+			it('should set error property info when passed as an argument', () => {
+				return expect(rpcGetPeersFailed)
+					.and.to.have.property('peerId')
+					.which.is.eql(peerId);
+			});
+
+			it('should set error property info when passed as an argument', () => {
+				expect(rpcGetPeersFailed)
+					.and.to.have.property('cause')
+					.and.to.be.a('function');
+
+				return expect(rpcGetPeersFailed.cause()).is.instanceOf(Error);
+			});
+
+			it('should set error property info when passed as an argument', () => {
```

Comment:
The two test cases have some overlap in coverage. The previous test case already checked that the rpcGetPeersFailed has a cause property which is a function. Also, it already checked that rpcGetPeersFailed.cause() is an instanceOf(Error). Maybe we should remove the first (less detailed) test case and only keep the more detailed one.

The test suite has to be maintained and is prone to change so at this stage of the project it's better to have fewer test cases which cover more application logic; that way we get more value out every test case.

Reply:
addressed